File: src/my_project/easy_problems/from1to50/length_last_word.py


if __name__ == '__main__':
    class Solution:
        def lengthOfLastWord(self, s: str) -> int:
            flag = False
            len_string = len(s)
            temp = 0
            if len_string == 0:
                return 0
            else:
                for i in range(len(s)):
                    flag = False
                    if s[i] == " ":
                        try:
                            s[i+1]
                        except:
                            return temp

                        for j in s[i:]:
                            if j != " ":
                                flag = True
                        if flag == False:
                            return temp
                        temp = 0
                        continue
                    temp += 1
                return temp



    solution = Solution()
    print(solution.lengthOfLastWord("Hello  World   "))

File: my_project_typescript/easy_problems_typescript/from1to50/lengthLastWord.ts
function lengthOfLastWord(s: string): number {
    // define length of the string
    s = s.trim();
    
    let stringLength: number = s.length;

    // initialize solution 
    let solution: number = 0;
    if (stringLength ===0){
        return 0;
    } else {
        stringLength = s.split(" ")[s.split(" ").length -1 ].length
        solution = stringLength
        return solution
    }
};


console.log(lengthOfLastWord("Hello World"));
console.log(lengthOfLastWord(" Hello"));
